components/sensor/bmp581.rst

```diff
@@ -0,0 +1,147 @@
+BMP581 Temperature+Pressure Sensor
+===========================================
+
+.. seo::
+    :description: Instructions for setting up BMP581 temperature and pressure sensors with ESPHome
+    :image: bmp581.jpg
+    :keywords: BMP581
+
+The ``bmp581`` sensor platform allows you to use your BMP581 
+(`datasheet <https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bmp581-ds004.pdf>`__, `SparkFun <https://www.sparkfun.com/products/20170>`__) temperature and pressure sensors with ESPHome. The :ref:`I²C <i2c>` bus is
+required to be set up in your configuration for this sensor to work.
+
+.. figure:: images/bmp581.jpg
+    :align: center
+    :width: 50.0%
+
+    BMP581 Temperature and Pressure Sensor.
+    (Credit: `SparkFun <https://www.sparkfun.com/products/20170>`__, image cropped and compressed)
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+      - platform: bmp581
+        temperature:
+          name: "Indoor Temperature"
+          iir_filter: 2x
+        pressure:
+          name: "Indoor Pressure"
+          oversampling: 128x
+        address: 0x47
+        update_interval: 60s
```

Comment:
Minimum only please, people need to read and understand the configuraton variables block below and configure to their needs.

```suggestion
    sensor:
      - platform: bmp581
        temperature:
          name: "Indoor Temperature"
        pressure:
          name: "Indoor Pressure"
```